lib/edge-runtime/requests.ts

```diff
@@ -3,10 +3,101 @@ import { fillNoChoiceRequiredParams } from "../actionUtils";
 import { Json } from "../database.types";
 import { ActionToHttpRequest } from "../models";
 import { Action } from "../types";
-import { deduplicateArray, filterKeys } from "../utils";
+import { deduplicateArray, filterKeys, swapKeysValues, isID } from "../utils";
 import { getHeader, getJsonMIMEType } from "./utils";
 import MediaTypeObject = OpenAPIV3_1.MediaTypeObject;
 
+export type IDStore = { [key: string]: string };
+
+export function removeIDs(
+  obj: Json | Json[],
+  existingStore?: IDStore,
+): {
+  cleanedObject: Json | Json[];
+  idStore: IDStore;
+} {
+  const store: IDStore = existingStore ?? {};
+  const removedObj = JSON.parse(JSON.stringify(obj));
+  let idIdx = 0;
+
+  function findAndReplaceID(json: Json | Json[]) {
+    if (!json || typeof json !== "object") return;
+
+    const entries = Array.isArray(json)
+      ? json.entries()
+      : Object.entries(json as { [key: string]: Json });
+
+    for (const [key, value] of entries) {
+      const k = key as string;
+      if (typeof value === "object") {
+        findAndReplaceID(value);
+      } else if (typeof value === "string") {
+        if (value.includes("/")) {
+          // URL-style string detected
+          const urlParts = value
+            .split("/")
+            .map((part) =>
+              typeof part === "string" && isID(part)
+                ? getOrGenerateID(part)
+                : part,
+            );
+          (json as any)[k] = urlParts.join("/");
+        } else if (isID(value)) {
+          (json as any)[k] = getOrGenerateID(value);
+        }
+      }
+    }
+  }
+
+  function getOrGenerateID(value: string): string {
+    let id = store[value];
+    if (!id) {
+      id = `ID${++idIdx}`;
```

Comment:
I think we want `{ID1}` so it's clearer to the AI that it's a variable

I think we also need a line in the prompt saying that variables like this in strings in function responses can be used in outputs like `{IDX}` - it 'knew' this by default for me, but worth testing this locally with fine-tuned 3.5 (I'll slack you with how to set up using fine-tuned 3.5 locally :smile:)

Reply:
I can test but do we need the AI to know that the id is a variable? If the AI thinks that the ID it needs to make an api call is "ID1" that's completly fine isn't it? Or  do you think this doesn't look like a real ID so will be confusing?